client-spark/common/src/main/java/org/apache/spark/shuffle/KryoSerializerWrapper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle;
+
+import java.nio.ByteBuffer;
+
+import scala.reflect.ClassTag$;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.serializer.KryoSerializer;
+import org.apache.spark.serializer.SerializerInstance;
+
+public class KryoSerializerWrapper {
```

Comment:
We shouldn't bind to KryoSeriliazerWrapper.

Reply:
I hope the shuffleHandleInfo could be shared by the driver and executors by seriliazed bytes.

Reply:
Can other RPCs also be implemented in this manner?

Reply:
Yes similar implementations could be acted in shuffleManager rpc service, because the driver and executor are always bounded into the same version, no compatiblity problems.

Reply:
This has been refactored to avoid de/serialization performance drop